STUDIO/org.openl.rules.webstudio/src/org/openl/rules/webstudio/web/CopyBean.java

```diff
@@ -466,6 +461,20 @@ public boolean isSupportsBranches() {
         return project != null && project.isSupportsBranches();
     }
 
+    public List<Repository> getAllowedRepositories() {
+        DesignTimeRepository designRepo = getUserWorkspace().getDesignTimeRepository();
+        return designRepo.getRepositories()
+            .stream()
+            .filter(repo -> !repo.supports()
+                .branches() || (repo.supports()
+                    .branches() && !((BranchRepository) repo).isBranchProtected(((BranchRepository) repo).getBranch())))

```

Comment:
Second check of `repo.supports().branches()` is unneccessary. If it false, we don't reach here (because we checked it before).